### PR TITLE
Remove Azure Pipelines

### PR DESCRIPTION
**Pull request by imbsky**
The test still failed, but it's not a problem with this PR itself.

### CODE REVIEW

**Comment by prometheansacrifice** [comment]
We rely on Azure for the prebuilts. I'm not aware of this being done on GH actions - has it been addressed?

**Comment by imbsky** [comment]
Not yet, but it will be possible soon.

**Comment by rgrinberg** [comment]
More to the point, are we running the integration tests with this pipeline?

**Comment by imbsky** [comment]
That's right. Also, automatic toolchain setup is currently disabled, and I remember that we have decided not to use Azure's artifact to distribute pre-built binaries. So we might be able to merge this, what do you think?

**Comment by rgrinberg** [comment]
Well, it’s more like that this package doesn’t produce any binaries so the publishing should be done outside this package. It doesn’t really matter if it’s done in azure or github actions. The integration tests still need to be be ported to CI however

**Comment by imbsky** [comment]
The integration test is already ported to Actions but is now broken regardless of this PR.
https://github.com/ocamllabs/vscode-ocaml-platform/pull/161/files#diff-f09ce9d24f2aeb928cee946c13324a06R54-R57

**Comment by rgrinberg** [comment]
I see. You need to clarify stuff like this in the description 🙂

Okay, so the pr indeed looks good. Let me just have a more in depth review later.

**Comment by imbsky** [comment]
As said in https://github.com/ocaml/dune/pull/3451, there are many benefits to migrating to Actions for CIs that need an OCaml environment. You will also benefit more in the future.

**Comment by imbsky** [comment]
> You need to clarify stuff like this in the description

Yes, indeed...

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -13,27 +13,50 @@ jobs:
           - macos-latest
           - ubuntu-latest
           - windows-latest
+        ocaml-version:
+          - 4.06.1
```
Comment (rgrinberg) [inline .github/workflows/workflow.yml:17]: Why 4.06?

Comment (imbsky) [reply .github/workflows/workflow.yml:17]: I thought we needed this version for testing with BuckleScript. Maybe it's not?

Comment (rgrinberg) [reply .github/workflows/workflow.yml:17]: Oh yes. Although I'm not sure we actually have any bucklescript tests at the moment.

**Comment by imbsky** [comment]
Actions VM doesn't have the global install directory of yarn in the path, so we will install esy with npm.

**Pull request merged**